internal/backend/runtime/omni/controllers/omni/schematic_configuration.go

```diff
@@ -217,3 +216,78 @@ func updateFinalizers(ctx context.Context, r controller.ReaderWriter, extensions
 
 	return r.AddFinalizer(ctx, extensions.Metadata(), SchematicConfigurationControllerName)
 }
+
+type machineExtensions struct {
+	machineStatus     *omni.MachineStatus
+	machineExtensions *omni.MachineExtensions
+	cluster           *omni.Cluster
+	extensionsList    []string
+}
+
+func newMachineExtensions(cluster *omni.Cluster, machineStatus *omni.MachineStatus, extensions *omni.MachineExtensions) (machineExtensions, error) {
+	me := machineExtensions{
+		machineStatus: machineStatus,
+		cluster:       cluster,
+	}
+
+	if extensions != nil && extensions.Metadata().Phase() == resource.PhaseRunning {
+		me.machineExtensions = extensions
+	}
+
+	detected, err := getDetectedExtensions(cluster, machineStatus)
+	if err != nil {
+		return me, err
+	}
+
+	me.extensionsList = append(me.extensionsList, detected...)
+
+	if me.machineExtensions != nil {
+		for _, e := range me.machineExtensions.TypedSpec().Value.Extensions {
+			if slices.Contains(me.extensionsList, e) {
+				continue
+			}
+
+			me.extensionsList = append(me.extensionsList, e)
```

Comment:
we probably want it general sort list of extensions, so that we normalize schematics like `a, b, c` and `b, a, c`

Reply:
I had the same idea initially, but then I remember you had the discussion with Utku that we shouldn't sort them.
As one extension can have an overlap with another one and we should allow users to change the order.

Reply:
I meant that on Image Factory level we probably can't as it changes the output, but our extensions shouldn't depend on the order they are added, so we can sort them to reduce the schematic bloat.